tests/unit/apps/test_subordinate.py

```diff
@@ -96,3 +96,56 @@ def test_channel_setter_invalid(status, channel):
 
     app.channel = channel
     assert app.channel == "ussuri/stable"
+
+
+@pytest.mark.parametrize(
+    "channel",
+    [
+        "focal/edge",
```

Comment:
I think those channels are a little bit misleading. IIRC, when charms comes from cs it always will be set to `stable`. E.g:

```yaml
applications:
  ubuntu:
    charm: cs:ubuntu-24
    series: bionic
    os: ubuntu
    charm-origin: charmstore
    charm-name: ubuntu
    charm-rev: 24
    charm-channel: stable
    exposed: false
```

So there is no need to have a parameterize function with different channels.

Reply:
I saw that conftest is alreasy setting the channel right as I said here

Reply:
made some improvements